src/ucs/sys/topo.c

```diff
@@ -29,22 +29,19 @@ typedef enum ucs_topo_kh_put {
 
 typedef struct ucs_topo_sys_dev_to_bus_arr {
     ucs_sys_bus_id_t bus_arr[UCS_TOPO_MAX_SYS_DEVICES];
-    unsigned count;
+    unsigned         count;
 } ucs_topo_sys_dev_to_bus_arr_t;
 
-KHASH_INIT(bus_to_sys_dev /*name*/,
-           ucs_bus_id_bit_rep_t /*key*/,
-           ucs_sys_device_t /*val*/,
-           1 /*is_map*/,
-           kh_int64_hash_func, kh_int64_hash_equal)
+KHASH_MAP_INIT_INT64(bus_to_sys_dev, ucs_sys_device_t);
 
-static khash_t(bus_to_sys_dev) ucs_topo_bus_to_sys_dev_hash;
-static ucs_recursive_spinlock_t  ucs_topo_bus_to_sys_dev_lock;
+typedef struct ucs_topo_global_ctx {
+    khash_t(bus_to_sys_dev)       bus_to_sys_dev_hash;
+    ucs_recursive_spinlock_t      bus_to_sys_dev_lock;
+    ucs_topo_sys_dev_to_bus_arr_t sys_dev_to_bus_lookup;
+    ucs_recursive_spinlock_t      sys_dev_to_bus_lock;
```

Comment:
can have one spinlock for both, for simplicity, and it doesn't have to be recursive

Reply:
Using one lock now. 

This would mean that concurrent use of `sys_dev_to_bus_lookup` array and `bus_to_sys_dev_hash` hashmap may see contention when different threads call `ucs_topo_find_device_by_bus_id` and `ucs_topo_get_distance`.